File: mobile/src/main/java/com/windscribe/mobile/upgradeactivity/UpgradeView.java
/*
 * Copyright (c) 2021 Windscribe Limited.
 */

package com.windscribe.mobile.upgradeactivity;

import androidx.annotation.Nullable;

import com.amazon.device.iap.model.Product;
import com.android.billingclient.api.Purchase;
import com.android.billingclient.api.SkuDetails;
import com.windscribe.mobile.upgradeactivity.UpgradeActivity.BillingType;
import com.windscribe.vpn.billing.WindscribeInAppProduct;

import java.util.List;

public interface UpgradeView {

    BillingType getBillingType();

    void getProducts(List<String> skuList);

    void goBackToMainActivity();

    void goToAddEmail();

    void goToConfirmEmail();

    void gotToClaimAccount();

    void hideProgressBar();

    boolean isBillingProcessFinished();

    void onPurchaseCancelled();

    //Called after onPurchaseUpdate returns successful
    void onPurchaseSuccessful(@Nullable List<Purchase> purchases);

    void querySkuDetails(List<String> products, boolean sub);

    void restorePurchase();

    void setBillingProcessStatus(boolean bProcessFinished);

    void setEmailStatus(boolean isEmailAdded, boolean isEmailConfirmed);

    void showBillingDialog(final WindscribeInAppProduct windscribeInAppProduct, boolean isEmailAdded,
            boolean isEmailConfirmed);

    void showBillingErrorDialog(String errorMessage);

    void showProgressBar(String message);

    void showToast(String toastText);

    void startPurchaseFlow(SkuDetails skuDetails, String accountID);

    void startPurchaseFlow(Product product);

    void startSignUpActivity();

    void startWindscribeActivity();

    void openUrlInBrowser(String url);
}

File: tv/src/main/java/com/windscribe/tv/upgrade/UpgradeView.kt
/*
 * Copyright (c) 2021 Windscribe Limited.
 */
package com.windscribe.tv.upgrade

import com.amazon.device.iap.model.Product
import com.android.billingclient.api.Purchase
import com.android.billingclient.api.SkuDetails
import com.windscribe.vpn.billing.WindscribeInAppProduct

interface UpgradeView {
    val billingType: UpgradeActivity.BillingType
    fun getProducts(skuList: List<String>)
    fun goBackToMainActivity()
    fun goToAddEmail()
    fun goToConfirmEmail()
    fun gotToClaimAccount()
    fun hideProgressBar()
    val isBillingProcessFinished: Boolean
    fun onPurchaseCancelled()
    fun onPurchaseSuccessful(purchases: List<Purchase>)
    fun querySkuDetails(products: List<String>, sub: Boolean)
    fun restorePurchase()
    fun setBillingProcessStatus(processFinished: Boolean)
    fun setEmailStatus(isEmailAdded: Boolean, isEmailConfirmed: Boolean)
    fun showBillingDialog(
        windscribeInAppProduct: WindscribeInAppProduct,
        isEmailAdded: Boolean,
        isEmailConfirmed: Boolean
    )
    fun showBillingErrorDialog(errorMessage: String)
    fun showProgressBar(message: String)
    fun showToast(toastText: String)
    fun startPurchaseFlow(skuDetails: SkuDetails, accountID: String?)
    fun startPurchaseFlow(product: Product)
    fun startSignUpActivity()
    fun startWindscribeActivity()
}
